client/dashboard/plugins/plugin/use-plugin.ts

```diff
@@ -15,8 +15,25 @@ export interface SiteWithPluginData extends Site {
 	isPluginActive: boolean;
 }
 
+/**
+ * Search for an icon on the cached marketplace search query results for a given plugin slug.
+ */
+const useMarketplaceSearchIcon = ( pluginSlug: string ) => {
+	const queryClient = useQueryClient();
+	const marketplaceSearchPluginData = queryClient
+		.getQueriesData< MarketplaceSearch >( {
+			queryKey: [ 'marketplace-search' ],
+			predicate: ( query ) => query.queryKey.includes( pluginSlug ),
+		} )
```

Comment:
Is there a good reason why we're not just using `useQuery()` here? It felt odd to have both a `queryKey` and `predicate`?